packages/mui-joy/src/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,306 @@
+ *
+ * API:
+ *
+ * - [Drawer API](https://mui.com/joy-ui/api/drawer/)
+ */
+const Drawer = React.forwardRef(function Drawer(inProps, ref) {
+  const props = useThemeProps<typeof inProps & { component?: React.ElementType }>({
+    props: inProps,
+    name: 'JoyDrawer',
+  });
+
+  const {
+    children,
+    anchor = 'left',
+    container,
+    disableAutoFocus = false,
+    disableEnforceFocus = false,
+    disableEscapeKeyDown = false,
+    disablePortal = false,
+    disableRestoreFocus = false,
+    disableScrollLock = false,
+    hideBackdrop = false,
+    onClose,
+    onKeyDown,
+    open,
+    component,
+    slots = {},
+    slotProps = {},
+    ...other
+  } = props;
+
+  const ownerState = {
+    ...props,
+    anchor,
+    disableAutoFocus,
+    disableEnforceFocus,
+    disableEscapeKeyDown,
+    disablePortal,
+    disableRestoreFocus,
+    disableScrollLock,
+    hideBackdrop,
+  };
+
+  const { getRootProps, getBackdropProps, rootRef, portalRef, isTopModal } = useModal({
+    ...ownerState,
+    rootRef: ref,
+  });
+
+  const classes = useUtilityClasses(ownerState);
+  const externalForwardedProps = { ...other, component, slots, slotProps };
+
+  const [SlotRoot, rootProps] = useSlot('root', {
+    ref: rootRef,
+    className: classes.root,
+    elementType: DrawerRoot,
+    externalForwardedProps,
+    getSlotProps: getRootProps,
+    ownerState,
+  });
+
+  const [SlotBackdrop, backdropProps] = useSlot('backdrop', {
+    className: classes.backdrop,
+    elementType: DrawerBackdrop,
+    externalForwardedProps,
+    getSlotProps: getBackdropProps,
+    ownerState,
+  });
+
+  const [SlotContent, contentProps] = useSlot('content', {
+    className: classes.content,
+    elementType: DrawerContent,
+    externalForwardedProps,
+    ownerState,
+  });
+
+  return (
+    <CloseModalContext.Provider value={onClose}>
+      <DrawerAnchorContext.Provider value={anchor}>
+        <DrawerOpenContext.Provider value={open}>
+          <Portal ref={portalRef} container={container} disablePortal={disablePortal}>
+            {/*
+             * Marking an element with the role presentation indicates to assistive technology
+             * that this element should be ignored; it exists to support the web application and
+             * is not meant for humans to interact with directly.
+             * https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md
+             */}
+            <SlotRoot {...rootProps}>
+              {!hideBackdrop ? <SlotBackdrop {...backdropProps} /> : null}
+              <FocusTrap
+                disableEnforceFocus={disableEnforceFocus}
+                disableAutoFocus={disableAutoFocus}
+                disableRestoreFocus={disableRestoreFocus}
+                isEnabled={isTopModal}
+                open={open}
+              >
+                <SlotContent {...contentProps} tabIndex={contentProps.tabIndex ?? -1}>
```

Comment:
This line is strange to me. Can it be included in the `additionalProps`?

```js
const [SlotContent, contentProps] = useSlot({
  …,
  additionalProps: {
    tabIndex: tabIndex ?? -1
  }
})
```

Reply:
Yep, refactored and added test for it.